android/src/main/java/com/swmansion/reanimated/nodes/StyleNode.java

```diff
@@ -21,23 +21,7 @@ public StyleNode(int nodeID, ReadableMap config, NodesManager nodesManager) {
   }
 
   @Override
-  protected WritableMap evaluate() {
-    JavaOnlyMap propMap = new JavaOnlyMap();
-    for (Map.Entry<String, Integer> entry : mMapping.entrySet()) {
-      Node node = mNodesManager.findNodeById(entry.getValue(), Node.class);
-      if (node instanceof TransformNode) {
-        propMap.putArray(entry.getKey(), (WritableArray) node.value());
-      } else {
-        Object val = node.value();
-        if (val instanceof Double) {
-          propMap.putDouble(entry.getKey(), (Double) val);
-        } else if (val instanceof String) {
-          propMap.putString(entry.getKey(), (String) val);
-        } else {
-          throw new IllegalStateException("Wrong style form");
-        }
-      }
-    }
-    return propMap;
+  protected Map evaluate() {
+    return mMapping;
```

Comment:
the code you removed it in fact necessary. Without it the input nodes evaluation for styles will only happen once. So when the styles updates as a result of running animations those changes won't be reflected on the view. You may not see the issue while testing maybe because you only rely on transforms, but if you update a prop that actually belongs to styles you should be able to see this (e.g. opacity). Also there is a chance you re-render from JS too often in which case the styles would get recreated.

Reply:
@kmagiera in the production app where I'm using this I am animating opacity in quite a few places, and there are also no JS re-renders on the page once it is loaded.

I've just moved the evaluation of the style input nodes to the PropsNode evaluate method

Reply:
@kmagiera I've just checked and actually in our use case we're animating width, opacity, fontSize and color too. All animate as expected.

Reply:
Great to hear it is working ok. However I am not sure I follow here. Are you saying you are testing this PR on top of some other changes fro mone of the PRs you listed?

Reply:
AH, ok, I think I understand this now. Please ignore my prev comment